CMake/macOSMaxAllowed.c

```diff
@@ -0,0 +1,8 @@
+#include <AvailabilityMacros.h>
```

Comment:
@MatthiasWM, @ManoloFLTK Please check if this header file is available on all targeted macOS versions. ISTR that the header `<Availability.h>` was used on older macOS versions.

Reply:
_Hmm, I'm confused, I thought I added this already:_

Maybe it's intended to fail if `<AvailabilityMacros.h>` does not exist? Then it's correct, obviously.

Reply:
`<Availability.h>` is the correct header, and `__MAC_OS_X_VERSION_MAX_ALLOWED` should always be defined there if we target OS X (vs. iPhone, watch, or what have you). If the include file does not exist, or the macro is not there, the caller falls back to the host system version `CURRENT_OSX_VERSION`.